### ISSUE TITLE
multiplex test failure

### ISSUE THREAD

**Issue by Anthchirp** [xia2/xia2#454]
to reproduce test failure run on linux Python2 or 3:
```
pytest --regression-full Test/regression/test_multiplex.py::test_proteinase_k Test/regression/test_multiplex.py::test_proteinase_k_filter_deltacchalf
```

**Comment by rjgildea** [comment]
Print statement added at beginning of multiplex, immediately after obtaining reflection tables from params.input.reflections:
```
print(reflections)
print([refl.get_flags(refl.flags.user_excluded_in_scaling).count(True) for refl in reflections])
```
Output from first multiplex test:
```
[<dials_array_family_flex_ext.reflection_table object at 0x7f68e3db31b8>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3e68>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3dc0>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3e30>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3ae8>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3538>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db31f0>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3928>]
[0, 0, 0, 0, 0, 0, 0, 0]
```
Output from second multiplex test:
```
[<dials_array_family_flex_ext.reflection_table object at 0x7f68e3db31b8>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3e68>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3dc0>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3e30>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3ae8>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3538>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db31f0>, <dials_array_family_flex_ext.reflection_table object at 0x7f68e3db3928>]
[8893, 0, 0, 0, 0, 0, 0, 0]
```
=> reflection table leaks between calls to multiplex.run() within the same process - command line option parser is returning the same physical objects that were used (and modified) in the previous pytest test

**Comment by Anthchirp** [comment]
First Jenkins build indicates this is now fixed. Will keep an eye on it.

**Comment by Anthchirp** [comment]
Subsequent test run failed, so this is still a thing

**Comment by Anthchirp** [comment]
Immediate issue seems to have been fixed by 7c1fa008fe389134c1f4b93dc6ba6509a9891575.
There may be remaining output dependence depending on `nproc` (ie. the test originally consistently failed on a 4-core machine, but consistently passed on a 12-core machine), but we won't be investigating this as part of this issue.
